### DISCUSSION TITLE
Embedding python in c++

### DISCUSSION THREAD

**Discussion by RoelVugts** [pybind/pybind11#4933] [Q&A]
Hi,

I have a c++ program that calls a python script using pybind11. I managed to get it working on my device (which has python installed). But when i try to run the program on another device (that does not have python installed) it gives an error saying: python311.dll is missing. I assumed that pybind11 completely embeds the python interpreter in the application. why does it still search for python311.dll then? am i doing something wrong?

OS: Windows

**Comment by sb5m** [comment]
If I am not mistaken python libs have to be present.

**Reply by RoelVugts** [reply]
As in have to be present on the device of the user or only present when compiling the program?

**Reply by sb5m** [reply]
As in have to be present, I would say. You can always give it a go and check, but I am quite certain the libs are needed.

**Comment by evguran** [comment]
While compiling you provide a python3xx.lib file
On running the app needs
python3xx.dll
python3xx.zip

For static compiling see this 
https://github.com/pybind/pybind11/issues/2678
